.github/workflows/upgrade-tests.yaml

```diff
@@ -0,0 +1,89 @@
+name: Upgrade Tests
+on: [push, pull_request, workflow_dispatch]
+
+jobs:
+  upgrade-tests:
+    runs-on: ubuntu-latest
+    timeout-minutes: 90
+    concurrency:
+      group: ${{ github.workflow }}-${{ github.ref }}
+      cancel-in-progress: true
+
+    env:
+      BRANCH_NAME: ${{ github.event.inputs.branch || 'master' }}
+
+    steps:
+      - name: Checkout noobaa-core (master/input branch)
+        uses: actions/checkout@v4
+        with:
+          repository: 'noobaa/noobaa-core'
+          path: 'noobaa-core'
+          ref: ${{ env.BRANCH_NAME }}
+
+      - name: Deploy minikube
+        run: |
+          cd ./noobaa-core
+          sudo bash ./.travis/deploy_minikube.sh
+
+      - name: Build noobaa
+        run: |
+          cd ./noobaa-core
+          make noobaa NOOBAA_TAG=noobaa-core:${{ env.BRANCH_NAME || 'master' }}
+
+      - name: Checkout noobaa-core (latest)
```

Comment:
I think "latest" might be confusing - as it is according to the branch that was set in the input, right?